docs/dev_building_web_assembly.rst

```diff
@@ -0,0 +1,146 @@
+
+  cd ..
+
+Then as before configure the library::
+
+  emcmake cmake -S libxml2 -B build-libxml2-release -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=usr/local -DZLIB_DIR=../usr/local/lib/cmake/ZLIB/ -DCMAKE_PREFIX_PATH=usr/local -DBUILD_SHARED_LIBS=OFF -DLIBXML2_WITH_ICONV=OFF -DLIBXML2_WITH_LZMA=OFF -DLIBXML2_WITH_PYTHON=OFF -DLIBXML2_WITH_TESTS=OFF -DLIBXML2_WITH_PROGRAMS=OFF
+
+There are seven additions to the configure command this time;
+
+1. We set ZLIB_DIR to enable the configuration to find our zlib library we just installed, -DZLIB_DIR=../usr/local/lib/cmake/ZLIB/.
+2. We turn off building a shared libxml2 library, -DBUILD_SHARED_LIBS=OFF.
+3. We turn off building libxml2 with iconv, -DLIBXML2_WITH_ICONV=OFF.
+4. We turn off building libxml2 with lzma compression, -DLIBXML2_WITH_LZMA=OFF.
+5. We turn off building libxml2 with Python, -DLIBXML2_WITH_PYTHON.
+6. We turn off building libxml2 with tests, -DLIBXML2_WITH_TESTS=OFF. and
+7. We turn off building libxml2 with programs, -DLIBXML2_WITH_PROGRAMS=OFF.
+
+As a general note it is better to specify the variable `ZLIB_DIR` as an absolute path and not a relative path as we have done here.
+Feel free to use the absolute path when you are configuring libXml2.
+
+Then as before we simply issue the build and install commands::
+
+  cd build-libxml2-release
+  make
+  make install
+
+That is all that is required for preparing the dependencies ready for libCellML.
+
+libCellML
+---------
+
+Following a similar approach to building the dependencies we can now configure and build libCellML.
+We don't install libCellML when building for web assembly, so there is no install command in the instructions.
+
+To get started we will make our current directory the `emscripten` directory we made at the start.
+Assuming we are in the directory where we left off from preparing the dependencies we simply change to the parent directory::
+
+  cd ..
+
+If you current directory is not where the dependency instructions left off then modify the command as appropriate for your situation.
+
+Okay, so let's get the source code using `git clone`::
+
+  git clone https://github.com/hsorby/libcellml.git -b emscripten
+
+The first thing you will note is that this code is not from the official libCellML repository.
+This work is still in a development stage and hasn't been properly vetted yet for general consumption.
+It is quite likely that you will encounter problems, and if you do don't be afraid to get in touch by creating an issue or whichever way you feel comfortable with.
+
+Now we will create a build directory::
+
+  mkdir build-libcellml-release
+
+We can now configure the library::
+
+  emcmake cmake -S libcellml -B build-libcellml-release -DLibXml2_DIR=../usr/local/lib/cmake/libxml2-2.9.10/ -DBUILD_TYPE=Release
+
+As with the earlier note it is best if `LibXml2_DIR` is an absolute path and not a relative path as shown here.
```

Comment:
As with the earlier note, it is best if `LibXml2_DIR` is an absolute path and not a relative path as shown here.

Reply:
9edec02